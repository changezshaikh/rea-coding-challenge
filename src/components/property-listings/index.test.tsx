import React from "react";
import { render } from "@testing-library/react";
import PropertyListings from ".";
import properties from "../../assets/data/properties.json";

describe("Property Listings tests", () => {
  test("should render without crashing and match snapshot", () => {
    const { container } = render(<PropertyListings {...properties} />);
    expect(container).toMatchSnapshot();
  });
});
