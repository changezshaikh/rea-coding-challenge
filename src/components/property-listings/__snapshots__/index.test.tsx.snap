// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Property Listings tests should render without crashing and match snapshot 1`] = `
<div>
  <section
    class="property-listings"
  >
    <div
      class="container"
    >
      <div
        class="column"
      >
        <div
          class="row"
        >
          <h2>
            Results
          </h2>
        </div>
        <div
          class="row"
        >
          <div
            class="property__container property-tile-result "
            data-testid="result-property-1"
          >
            <div
              class="property-header"
              style="background-color: rgb(255, 229, 18);"
            >
              <img
                alt="Agency Logo"
                class="logo-image"
                src="https://i1.au.reastatic.net/170x32/d9e65c666e427e655fb63dcfe73f50d4ac7ff9a58c173db9474bd92e75b01070/main.gif"
              />
            </div>
            <img
              alt="Property"
              class="property-image"
              src="http://i2.au.reastatic.net/640x480/20bfc8668a30e8cabf045a1cd54814a9042fc715a8be683ba196898333d68cec/main.jpg"
            />
            <div
              class="property-footer"
            >
              <span
                class="property-price"
              >
                $726,500
              </span>
              <button
                type="button"
              >
                Disable
              </button>
              <button
                class="property-button "
                type="button"
              >
                Save
              </button>
            </div>
          </div>
          <div
            class="property__container property-tile-result "
            data-testid="result-property-2"
          >
            <div
              class="property-header"
              style="background-color: rgb(252, 250, 59);"
            >
              <img
                alt="Agency Logo"
                class="logo-image"
                src="https://i4.au.reastatic.net/170x32/a3bd69c1a5c651eca02dd829fdd4d11d1a0b3bb6b00db79410817d17067bd495/main.gif"
              />
            </div>
            <img
              alt="Property"
              class="property-image"
              src="http://i1.au.reastatic.net/640x480/88586227f9176f602d5c19cf06261108dbb29f03e30d1c4ce9fc2b51fb1e4bd6/main.jpg"
            />
            <div
              class="property-footer"
            >
              <span
                class="property-price"
              >
                $560,520
              </span>
              <button
                type="button"
              >
                Disable
              </button>
              <button
                class="property-button "
                type="button"
              >
                Save
              </button>
            </div>
          </div>
          <div
            class="property__container property-tile-result "
            data-testid="result-property-3"
          >
            <div
              class="property-header"
              style="background-color: rgb(87, 181, 224);"
            >
              <img
                alt="Agency Logo"
                class="logo-image"
                src="https://i1.au.reastatic.net/170x32/b269b079bf554d2ae9ca01d578bb3d80ec5fbb4f57062bbfcd51bbf1f2d13981/main.gif"
              />
            </div>
            <img
              alt="Property"
              class="property-image"
              src="http://i4.au.reastatic.net/640x480/98cee1b2a3a64329921fc38f7e2926a78d41fcc683fc48fb8a8ef2999b14c027/main.jpg"
            />
            <div
              class="property-footer"
            >
              <span
                class="property-price"
              >
                $826,500
              </span>
              <button
                type="button"
              >
                Disable
              </button>
              <button
                class="property-button "
                type="button"
              >
                Save
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="column"
      >
        <div
          class="row"
        >
          <h2>
            Saved properties
          </h2>
        </div>
        <div
          class="row"
        >
          <div
            class="property__container property-tile-saved "
            data-testid="saved-property-4"
          >
            <div
              class="property-header"
              style="background-color: rgb(0, 0, 0);"
            >
              <img
                alt="Agency Logo"
                class="logo-image"
                src="https://i2.au.reastatic.net/170x32/3015ba9710c7e3ddc2ac30f45fd7906df5b04e442a7f6948f75a6029b8b871e2/main.gif"
              />
            </div>
            <img
              alt="Property"
              class="property-image"
              src="http://i2.au.reastatic.net/640x480/5e84d96722dda3ea2a084d6935677f64872d1d760562d530c3cabfcb7bcda9c2/main.jpg"
            />
            <div
              class="property-footer"
            >
              <span
                class="property-price"
              >
                $526,500
              </span>
              <button
                class="property-button property-button--saved"
                type="button"
              >
                Remove
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;
